api/proto/teleport/legacy/types/events/events.proto

```diff
@@ -7950,3 +7953,97 @@ message UserLoginAccessListInvalid {
     (gogoproto.jsontag) = ""
   ];
 }
+
+// AWSICResourceSyncEvent defines import and reconciliation event data for a
+// account, account assignment, permission set or user group type.
+message AWSICResourceSyncEvent {
+  //  Resource represents a single user group, account, account assignment or permission set.
+  message Resource {
+    // ID is an ID of the resource.
+    string id = 1 [(gogoproto.jsontag) = "id, omitempty"];
+    // Name is a display name of the resource.
+    string name = 2 [(gogoproto.jsontag) = "name, omitempty"];
+    // ARN is the ARN of the resource.
+    string arn = 3 [(gogoproto.jsontag) = "arn,omitempty"];
+    // AssignedPermissionSetARN holds the ARN value of the assigned permission set.
+    // Only used for an account assignment resource.
+    string assignedPermissionSetARN = 4 [(gogoproto.jsontag) = "assigned_permission_set_arn, omitempty"];
+  }
+
+  // Array of imported resource object.
+  repeated Resource items = 2 [(gogoproto.jsontag) = "items,omitempty"];
+  // TotalItems records total number of imported resources.
+  int32 total_items = 3 [(gogoproto.jsontag) = "total_items,omitempty"];
+  // Status is a resource sync status.
+  Status Status = 4 [
+    (gogoproto.nullable) = false,
+    (gogoproto.embed) = true,
+    (gogoproto.jsontag) = ""
+  ];
+}
+
+// AWSICResourceSync is emitted when the AWS Identity Center integration
+// imports resource from Identity Center and reconcile them in Teleport. The reconciliation
+// happens individually but we batch them and submit the event at once.
+message AWSICResourceSync {
+  // Metadata is the event metadata.
+  Metadata metadata = 1 [
+    (gogoproto.nullable) = false,
+    (gogoproto.embed) = true,
+    (gogoproto.jsontag) = ""
+  ];
+
+  //  Account is the AWS account.
+  AWSICResourceSyncEvent account = 2 [(gogoproto.jsontag) = "account,omitempty"];
+  //  AccountAssignment is a permission assignment record bound to AWS account and permission set.
+  AWSICResourceSyncEvent account_assignment = 3 [(gogoproto.jsontag) = "account_assignment,omitempty"];
+  //  UserGroup is a user group available in the AWS IAM Identity Center instance.
+  AWSICResourceSyncEvent user_group = 4 [(gogoproto.jsontag) = "user_group,omitempty"];
+  //  PermissionSet is the permission set configured in the AWS IAM Identity Center instance.
+  AWSICResourceSyncEvent permission_set = 5 [(gogoproto.jsontag) = "permission_set,omitempty"];
```

Comment:
I'm trying to imagine how this type will evolve over time. Instead of 4 different fields with lists of synced resources, could it be easier to have 1 list of all synced resources? then we wouldn't need to change these even proto types if we start syncing new types of resources. just a thought